File: lib/src/main/java/net/kibotu/android/bloodhound/internal/ScreenTracker.java
package net.kibotu.android.bloodhound.internal;

import android.support.annotation.NonNull;
import com.google.android.gms.analytics.HitBuilders;
import com.google.android.gms.analytics.Tracker;

public final class ScreenTracker {

    private final String screenName;
    private final Tracker tracker;
    @NonNull
    private final HitBuilders.ScreenViewBuilder builder;

    private ScreenTracker(final Tracker tracker, final String screenName) {
        this.tracker = tracker;
        this.screenName = screenName;
        this.builder = new HitBuilders.ScreenViewBuilder();
    }

    @NonNull
    public static ScreenTracker from(final Tracker tracker, final String screenName) {
        return new ScreenTracker(tracker, screenName);
    }

    public void track() {
        tracker.setScreenName(screenName);
        tracker.send(builder.build());
        tracker.setScreenName(null);
    }

    public void setNewSession() {
        builder.setNewSession();
    }
}

File: lib/src/main/java/net.kibotu/bloodhound/internal/ScreenTracker.kt
/*
 * Created by [Jan Rabe](https://about.me/janrabe)
 */

package net.kibotu.bloodhound.internal

import com.google.android.gms.analytics.HitBuilders
import com.google.android.gms.analytics.Tracker

internal class ScreenTracker(private val tracker: Tracker, private val screenName: String) {

    private val builder: HitBuilders.ScreenViewBuilder = HitBuilders.ScreenViewBuilder()

    fun track() {
        tracker.setScreenName(screenName)
        tracker.send(builder.build())
        tracker.setScreenName(null)
    }

    fun setNewSession() {
        builder.setNewSession()
    }
}